app/lib/bk/compat/parsers/bitrise/slack.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+module BK
+  module Compat
+    module BitriseSteps
+      # Handles the translation of slack notifications
+      class Translator
+        def translate_slack(inputs)
+          warnings = generate_warnings(inputs)
+
+          if error_config?(inputs)
+            create_dual_notifications(inputs, warnings)
+          else
+            create_single_notification(inputs, warnings)
+          end
+        end
+
+        private
+
+        def error_config?(inputs)
+          inputs.keys.any? { |k| k.end_with?('_on_error') && !inputs[k].to_s.empty? }
+        end
+
+        def generate_warnings(inputs)
+          supported_options = %w[
+            channel channel_on_error
+            text text_on_error
+            webhook_url webhook_url_on_error
+            emoji emoji_on_error
+            from_username from_username_on_error
+            color color_on_error
+            title title_on_error
+            message message_on_error
+          ]
+
+          unsupported = inputs.keys.reject { |key| supported_options.include?(key) }
```

Comment:
it is a lot cleaner to use array differences here:

```suggestion
          unsupported = supported_options.difference(inputs.keys)
```